### PR TITLE
Add source support beyond 'ulb' and 'obs'

### PR DESCRIPTION
**Pull request by PurpleGuitar**
Changes in this pull request:
- Remove restriction so that any source can have resources, not just "ulb" and "obs"
- Add support for developing using a Docker container
- Update version to 1.0.5 b5

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+rebuild:
+	# Re-builds image from scratch (takes a while!)
+	docker build --no-cache --tag bw-local-dev .
+
+run:
+	# Starts BTT-Writer in a container.  The user directory is in a volume.
+	# Runs detached to prevent messing up the tty.  (Use `make stop` to end if needed.)
+	test $(DISPLAY) # If blank, then $$DISPLAY is not set
+	test ! $(CONTAINER_ID) # If not blank, then container is already running
+	docker run --detach --rm --volume $(VOLUME_LABEL):/root --env DISPLAY="${DISPLAY}" $(IMAGE_LABEL)
```
Comment (danparisd) [inline makefile:25]: I needed this to get it to run on native linux:
```suggestion
	docker run --detach --rm --net=host --volume="${HOME}/.Xauthority:/root/.Xauthority:rw" --volume $(VOLUME_LABEL):/root --env DISPLAY="${DISPLAY}" $(IMAGE_LABEL)
```

Comment (PurpleGuitar) [reply makefile:25]: OK, cool -- I'll check and see if that breaks WSL...

Comment (PurpleGuitar) [reply makefile:25]: Seems to work fine in WSL.

**Comment by PurpleGuitar** [comment]
> LGTM.
> 
> You may want a wsl run command and a separate native linux run command? And maybe a comment as to your setup on wsl to get it running.

I updated the docs to let WSL users know they need to point to their X display manager.

**Pull request merged**